src/arrayMethodReduce.js

```diff
@@ -5,7 +5,19 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let previousValue = initialValue;
+    let startWith = 0;
+
+    if (arguments.length < 2) {
+      previousValue = this[0];
+      startWith = 1;
+    }
+
+    for (let i = startWith; i < this.length; i++) {
```

Comment:
This is about English not code

it is better to use 'startFrom', 'startAt' or 'startIndex'  instead of 'startWith'
check the difference between them